editor/src/components/canvas/remix/utopia-remix-root-component.tsx

```diff
@@ -1,66 +1,116 @@
+import { RemixContext } from '@remix-run/react/dist/components'
+import type { RouteModules } from '@remix-run/react/dist/routeModules'
 import React from 'react'
-import { UNSAFE_RemixContext as RemixContext } from '@remix-run/react'
-import { DefaultFutureConfig } from './remix-utils'
-import {
-  RemixErrorBoundary,
-  V2_RemixRootDefaultErrorBoundary,
-} from '@remix-run/react/dist/errorBoundaries'
-import { RouterProvider, createMemoryRouter } from 'react-router'
-
-// TODO temporary, we need the real manifest
-const dummyManifest = {
-  entry: {
-    imports: [],
-    module: '',
-  },
-  routes: {
-    root: {
-      hasAction: false,
-      hasCatchBoundary: false,
-      hasErrorBoundary: false,
-      hasLoader: false,
-      id: 'root',
-      imports: [],
-      module: '/',
-      path: '',
+import { createMemoryRouter, RouterProvider } from 'react-router'
+import { UTOPIA_PATH_KEY } from '../../../core/model/utopia-constants'
+import type { ElementPath } from '../../../core/shared/project-file-types'
+import { useRefEditorState, useEditorState, Substores } from '../../editor/store/store-hook'
+import { getDefaultExportNameAndUidFromFile } from './remix-utils'
+import * as EP from '../../../core/shared/element-path'
+import { PathPropHOC } from './path-props-hoc'
+
+function useGetRouteModules(basePath: ElementPath) {
+  const remixDerivedDataRef = useRefEditorState((store) => store.derived.remixData)
+  const projectContentsRef = useRefEditorState((store) => store.editor.projectContents)
+
+  const defaultExports = useEditorState(
+    Substores.derived,
+    (store) => {
+      const routeModuleCreators = store.derived.remixData?.routeModuleCreators ?? {}
+      return Object.values(routeModuleCreators).map(
+        (rmc) => getDefaultExportNameAndUidFromFile(projectContentsRef.current, rmc.filePath)?.name,
+      )
     },
-  },
-  url: '',
-  version: '',
+    '',
+  )
+
+  return React.useMemo(() => {
+    const defaultExportsIgnored = defaultExports // Forcibly update the routeModules only when the default exports have changed
+
+    if (remixDerivedDataRef.current == null) {
+      return null
+    }
+
+    const routeModulesResult: RouteModules = {}
+    for (const [routeId, value] of Object.entries(
+      remixDerivedDataRef.current.routeModuleCreators,
+    )) {
+      const nameAndUid = getDefaultExportNameAndUidFromFile(
+        projectContentsRef.current,
+        value.filePath,
+      )
+      if (nameAndUid == null) {
+        continue
+      }
+
+      const relativePath =
+        remixDerivedDataRef.current.routeModulesToRelativePaths[routeId].relativePath
+
+      const defaultComponent = (componentProps: any) =>
+        value
+          .executionScopeCreator(projectContentsRef.current)
+          .scope[nameAndUid.name]?.(componentProps) ?? <React.Fragment />
```

Comment:
Why a Fragment?

Reply:
We added a task for this to the paper doc for later.